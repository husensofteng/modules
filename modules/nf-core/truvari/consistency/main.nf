
process TRUVARI_CONSISTENCY {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/truvari:4.1.0--pyhdfd78af_0':
        'biocontainers/truvari:4.1.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(vcf1), path(vcf2), path(vcf3)

    output:
    tuple val(meta), path("*.consistency") , emit: consistency
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def vcf3   = vcf3 ?: "$vcf3"

    """
    truvari \\
        consistency \\
        $args \\
        $vcf1 \\
        $vcf2  \\
        $vcf3 > ${prefix}.consistency

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        truvari: \$(echo \$(truvari version 2>&1) | sed 's/^Truvari v//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.consistency

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        truvari: \$(echo \$(truvari version 2>&1) | sed 's/^Truvari v//' ))
    END_VERSIONS
    """
}
