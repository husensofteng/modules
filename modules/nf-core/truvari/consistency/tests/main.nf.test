nextflow_process {

    name "Test Process TRUVARI_CONSISTENCY"
    script "../main.nf"
    process "TRUVARI_CONSISTENCY"

    tag "modules"
    tag "modules_nfcore"
    tag "truvari"
    tag "truvari/consistency"

    test("homo_sapiens - illumina - two vcf") {

        when {
            process {
                """
                input = [
                        [ id:'test' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_test2_paired_mutect2_calls_vcf_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test2_haplotc_vcf_gz'], checkIfExists: true),
                        []
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.txt).match("txt") }
            )
        }

    }

}
