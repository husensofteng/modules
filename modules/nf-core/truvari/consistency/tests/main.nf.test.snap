{
    "homo_sapiens - illumina - two vcf": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.json:md5,6413b4e25c59d9039e76104c38e8ad78"
                ]
            ],
            [
                "versions.yml:md5,2d1bbe95825a39d1c96c609fc5a1bb36"
            ]
        ],
        "timestamp": "2023-12-14T10:46:38.575425036"
    },
    "homo_sapiens - illumina - three vcf": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.json:md5,6eb1d42e8e0ae82a48c2dea1fed210f6"
                ]
            ],
            [
                "versions.yml:md5,2d1bbe95825a39d1c96c609fc5a1bb36"
            ]
        ],
        "timestamp": "2023-12-14T10:46:55.53173233"
    }
}